wp-includes/service-workers.php

```diff
@@ -126,13 +126,43 @@ function wp_print_service_workers() {
 	?>
 	<script>
 		if ( navigator.serviceWorker ) {
-			window.addEventListener('load', function() {
+			var updatedSw;
+
+			window.addEventListener( 'load', function() {
 				<?php foreach ( $scopes as $name => $scope ) { ?>
 					navigator.serviceWorker.register(
 						<?php echo wp_json_encode( wp_get_service_worker_url( $name ) ); ?>,
 						<?php echo wp_json_encode( compact( 'scope' ) ); ?>
-					);
+					).then( reg => {
+						reg.addEventListener( 'updatefound', () => {
+							updatedSw = reg.installing;
+							updatedSw.addEventListener( 'statechange', () => {
+
+								// If new service worker is available, show notification.
+								if ( 'installed' === updatedSw.state ) {
+									if ( navigator.serviceWorker.controller ) {
+
+										// Allows opting out from updating the service worker automatically.
+										if ( wp.hooks.applyFilters( 'pwa.serviceWorker.skipWaiting', true ) ) {
+											updatedSw.postMessage( { action: 'skipWaiting' } );
```

Comment:
I see this being useful if the user clicks on the update link. But the PHP filter `wp_service_worker_skip_waiting` filter should manage whether or not the notification is displayed at all, right? If the filter returns true then it should just `skipWaiting` whenever there is an update?